pkg/wharfapi/branch.go

```diff
@@ -1,70 +1,39 @@
 package wharfapi
 
 import (
-	"encoding/json"
 	"fmt"
-	"net/http"
-)
-
-// Branch holds metadata about a Git branch for a given Wharf project.
-type Branch struct {
-	BranchID  uint   `json:"branchId"`
-	ProjectID uint   `json:"projectId"`
-	Name      string `json:"name"`
-	TokenID   uint   `json:"tokenId"`
-	Default   bool   `json:"default"`
-}
-
-// PutBranch tries to match an existing branch by ID or by name,
-// or adds a new a provider if none matched, by invoking the HTTP request:
-// 	PUT /api/branch
-func (c Client) PutBranch(branch Branch) (Branch, error) {
-	newBranch := Branch{}
-
-	body, err := json.Marshal(branch)
-	if err != nil {
-		return newBranch, err
-	}
-
-	url := fmt.Sprintf("%s/api/branch", c.APIURL)
-	ioBody, err := doRequest("POST | BRANCH |", http.MethodPost, url, body, c.AuthHeader)
-	if err != nil {
-		return newBranch, err
-	}
 
-	defer (*ioBody).Close()
-
-	err = json.NewDecoder(*ioBody).Decode(&newBranch)
-	if err != nil {
-		return newBranch, err
-	}
+	"github.com/iver-wharf/wharf-api/pkg/model/request"
+	"github.com/iver-wharf/wharf-api/pkg/model/response"
+)
 
-	return newBranch, nil
+// CreateProjectBranch adds a branch to the project with the matching
+// project ID by invoking the HTTP request:
+//  POST /api/project/{projectId}/branch
+func (c Client) CreateProjectBranch(projectID uint, branch request.Branch) (response.Branch, error) {
+	var newBranch response.Branch
+	path := fmt.Sprintf("/api/project/%d/branch", projectID)
+	err := c.PostJSONUnmarshal(path, nil, &branch, &newBranch)
```

Comment:
Don't need to take the address of the request body (`&branch`). It could possibly give a tiny bit of performance and decrease memory allocations, but with it I find it more difficult to quickly see what's getting populated by the unmarshal and what's just another parameter, and the structs aren't THAT heavy anyway.